lib/komponent/core/component_path_resolver.rb

```diff
@@ -0,0 +1,66 @@
+module Komponent
+  class ComponentPathResolver
+    attr_accessor :path
+
+    def initialize(component)
+      @component = component
+      resolve
+    end
+
+    protected
+
+    def resolve
+      component_found = nil
+
+      component_paths.each do |component_path|
+        component_found = find_component(component_path)
+        break if component_found
+      end
+
+      raise ComponentPathResolver::MissingComponentError.new(component_paths) unless component_found
+
+      component_parts.each do |part|
+        component_found += part
+      end
```

Comment:
We can shorten that to something like `component_found = component_found.join(@component)` (or `+=` if you prefer this form)